R/generate_eval_data.R

```diff
@@ -0,0 +1,110 @@
+#' from the list of targets in the config, with properties "target_id",
+#' "metrics", and "disaggregate_by".
+#'
+#' @noRd
+generate_target_eval_data <- function(hub_path,
+                                      config,
+                                      out_path,
+                                      oracle_output,
+                                      target) {
+  target_id <- target$target_id
+  metrics <- target$metrics
+  disaggregate_by <- target$disaggregate_by
+  eval_windows <- config$eval_windows
+
+  task_groups_w_target <- get_task_groups_w_target(hub_path, target_id)
+  metric_name_to_output_type <- get_metric_name_to_output_type(task_groups_w_target, metrics)
+
+  for (eval_window in eval_windows) {
+    model_out_tbl <- load_model_out_in_window(hub_path, target$target_id, eval_window)
+
+    # overall scores, not disaggregated by a task id variable
+    get_and_save_scores(
+      model_out_tbl = model_out_tbl,
+      oracle_output = oracle_output,
+      metric_name_to_output_type = metric_name_to_output_type,
+      target_id = target_id,
+      window_name = eval_window$window_name,
+      by = NULL,
+      out_path = out_path
+    )
+
+    # scores disaggregated by a task id variable
+    for (by in disaggregate_by) {
+      get_and_save_scores(
+        model_out_tbl = model_out_tbl,
+        oracle_output = oracle_output,
+        metric_name_to_output_type = metric_name_to_output_type,
+        target_id = target_id,
+        window_name = eval_window$window_name,
+        by = by,
+        out_path = out_path
+      )
+    }
+  }
+}
+
+
+#' Get and save scores for a target in a given evaluation window,
+#' collecting across different output types as necessary.
+#' Scores are saved in .csv files in subdirectorys under out_path with one of
+#' two structures:
+#' - If by is NULL, the scores are saved in
+#' out_path/target_id/window_name/scores.csv
+#' - If by is not NULL, the scores are saved in
+#' out_path/target_id/window_name/by/scores.csv
+#' @noRd
+get_and_save_scores <- function(model_out_tbl, oracle_output, metric_name_to_output_type,
+                                target_id, window_name, by,
+                                out_path) {
+  # Iterate over the output types and calculate scores for each
+  scores <- purrr::map(
+    unique(metric_name_to_output_type$output_type),
+    ~ hubEvals::score_model_out(
+      model_out_tbl = model_out_tbl |> dplyr::filter(output_type == !!.x),
+      oracle_output = oracle_output,
+      metrics = metric_name_to_output_type$metric[
+        metric_name_to_output_type$output_type == .x
+      ],
+      by = c("model_id", by)
+    )
+  ) |>
+    purrr::reduce(dplyr::left_join, by = c("model_id", by))
+
+  if (is.null(by)) {
+    target_window_by_out_path <- file.path(out_path, target_id, window_name)
+  } else {
+    target_window_by_out_path <- file.path(out_path, target_id, window_name, by)
+  }
```

Comment:
avoiding repetition.


```suggestion
  target_window_by_out_path <- file.path(out_path, target_id, window_name)
  if (!is.null(by)) {
    target_window_by_out_path <- file.path(target_window_out_path, by)
  }
```